lib/plugins/aws/package/compile/events/schedule/index.js

```diff
@@ -14,22 +17,67 @@ class AwsCompileScheduledEvents {
       'package:compileEvents': this.compileScheduledEvents.bind(this),
     };
 
-    // TODO: Complete schema, see https://github.com/serverless/serverless/issues/8032
     this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'schedule', {
-      anyOf: [{ type: 'string' }, { type: 'object' }],
+      oneOf: [
+        { type: 'string', pattern: scheduleSyntax },
+        {
+          type: 'object',
+          properties: {
+            rate: { type: 'string', pattern: scheduleSyntax },
+            enabled: { type: 'boolean' },
+            name: { type: 'string', minLength: 1, maxLength: 64, pattern: '[\\.\\-_A-Za-z0-9]+' },
+            description: { type: 'string', maxLength: 512 },
+            input: {
+              oneOf: [
+                { type: 'string', maxLength: 8192, pattern: jsonPattern },
+                {
+                  type: 'object',
+                  properties: {
+                    body: { type: 'string', pattern: jsonPattern },
+                  },
+                },
+              ],
+            },
+            inputPath: { type: 'string', maxLength: 256 },
+            inputTransformer: {
+              type: 'object',
+              properties: {
+                inputTemplate: {
+                  type: 'string',
+                  minLength: 1,
+                  maxLength: 8192,
+                  pattern: jsonPattern,
+                },
+                inputPathsMap: { type: 'object' },
+              },
+              required: ['inputTemplate'],
+              additionalProperties: false,
+            },
+          },
+          required: ['rate'],
+          dependencies: {
+            input: {
+              allOf: [
+                { not: { required: ['inputPath'] } },
+                { not: { required: ['inputTransformer'] } },
+              ],
+            },
+            inputPath: {
+              allOf: [
+                { not: { required: ['input'] } },
+                { not: { required: ['inputTransformer'] } },
+              ],
+            },
+            inputTransformer: {
+              allOf: [{ not: { required: ['input'] } }, { not: { required: ['inputPath'] } }],
+            },
```

Comment:
While this schema works, I think it'll present user with cryptic misleading error message (something as `functions.foo.events[1].schedule should NOT be valid`)

We have a [place](https://github.com/serverless/serverless/blob/5f85543d8e47a9afaec6811a57e1235e82949550/lib/classes/ConfigSchemaHandler/normalizeAjvErrors.js#L208-L239) where we can add hooks to improve such messages, but I think it'll be hard to configure a generic meaningful message that'll work well for all `not` validation cases (at least it's hard for me to come up with any) (?)

In light of that I think it's probably good to validate this case as we validated so far (with inline validation) or just leave this error to surface from AWS (I'm pretty sure what they report it's also meaningful)

What do you think?

Reply:
Agreed, it's unfortunate it's not easy to add custom error messages.  I did look into this, but it would involve quite a lot of change.

How about this: if this validation is primarily intended to prevent silent errors, would it make sense to do basic validation here and defer to AWS for anything they're validating.  For this case, I would remove the mutual exclusion validation here and turn the JSON validation into a basic string type and length check.  Does that fit with what your expectation of this validation is for?

Reply:
@andyjduncan yes I totally agree with that. Let's remove validation for mutual exclusion from both schema and inline logic, so it's purely in hands of AWS.

Reply:
@medikoo I've updated this to limit the validation to type and length checks, basic patterns, and limiting properties where appropriate.  JSON validation and input property combinations will be validated by AWS.